### DISCUSSION TITLE
WingetUI 3.0 Alpha 0

### DISCUSSION THREAD

**Discussion by marticliment** [Devolutions/UniGetUI#1782] [General]
This is the first release of the brand-new WingetUI, based on C#.NET + WinUI3.
This release contains the base WingetUI application, completely rewritten from scratch.
### **However, the following features are NOT available yet:**
 - Importing and exporting packages
 - Installed package backups
 - Package tags
 - Adding and removing sources (installed sources are listed)
 - The API that communicates with WingetUI Widgets
 - The API that allows WingetUI to open when a package was shared to you (you can still share packages)
 - Viewing package details (The interface works, but it will show no data).
 - WingetUI ran as administrator will crash when opening a File/Folder picker
 - The built-in WingetUI autoupdater. **You may want to subscribe to this repository to get notified when a new release is available.**

The rest of features are expected to work as they did on WingetUI 2.2.0 and prior. 
This beta release may be unstable. It is not intended to be daily-driven.

<hr /><em>This discussion was created from the release <a href='https://github.com/marticliment/WingetUI/releases/tag/3.0.0-alpha0'>WingetUI 3.0 Alpha 0</a>.</em>

**Comment by Spacellary** [comment]
**Huge!**

**Comment by Otiel** [comment]
I'm curious, why the rewrite in .NET?

**Reply by meenbeese** [reply]
Python does not scale well, I would guess

**Reply by marticliment** [reply]
Deploying desktop aplications is hard, due to Python's interpreted nature.
Interface lagged and had little to none accessibility (It was very difficult to implement with Qt), and Python's slowlyness was noticeable when batching operations.
Also, .NET is more integrated with Windows, giving advantages, for example, when spawning processes.
You can see the full list of reasons on #1724

**Reply by Otiel** [reply]
Thanks for the reply! 

Are you already proficient in C# or learning as you go? 😊

**Reply by marticliment** [reply]
Learning as I go

**Reply by Otiel** [reply]
👍 don't hesitate to ask if you need some help.

**Reply by marticliment** [reply]
Will do, thank you!

**Comment by raghavdhingra24** [comment]
~~Please try to provide a portable version of this to allow trying without breaking the existing installation.~~

Edit Just noticed, it was in the installer. Sorry for posting unnecessaily

**Comment by Rakstrooper** [comment]
Hi Does this integrate all sources/indexes that were in DUMO and SUMO? thanks!
And I notice it asks to install windows app SDK stable, wondering since it is an alpha if using preview versions of windows app SDK (1.5) would accelerate its development goals.

**Reply by marticliment** [reply]
From DUMo not, WingetUI des not provide driver updates.

Regarding SUMo, WingetUI uses various package managers to get the software, so most likely yes.

**Comment by marticliment** [comment]
A second alpha release is available to download from here: https://github.com/marticliment/WingetUI/releases/tag/3.0.0-alpha1
